packages/stargate/src/stargateclient.ts

```diff
@@ -372,9 +372,9 @@ export class StargateClient {
 
     const broadcasted = await this.forceGetTmClient().broadcastTxSync({ tx });
     if (broadcasted.code) {
-      throw new Error(
-        `Broadcasting transaction failed with code ${broadcasted.code} (codespace: ${broadcasted.codeSpace}). Log: ${broadcasted.log}`,
-      );
+      return new Promise((resolve, reject) => {
+        reject(new Error(`Broadcasting transaction failed with code ${broadcasted.code} (codespace: ${broadcasted.codeSpace}). Log: ${broadcasted.log}`));
+      });
```

Comment:
Can you use `return Promise.reject(new Error(…` instead? Otherwise 👌

Reply:
@webmaster128  I've updated (and squashed) my PR.